src/pool/srv_target.c

```diff
@@ -754,6 +784,12 @@ ds_pool_stop(uuid_t uuid)
 	pool->sp_stopping = 1;
 
 	ds_iv_ns_stop(pool->sp_iv_ns);
+
+	/* Though all containers started in pool_alloc_ref, we need stop all
+	 * containers service before tgt_ec_eqh_query_ult(), otherwise container
+	 * EC aggregation ULT might try to access ec_eqh_query structure.
```

Comment:
What confused me is that: The 'cont_ec_eph' is allocated and inserted in the global pool list by ds_cont_ec_eph_insert() on EC aggregation ULT start, but it's freed by cont_ec_eph_destroy() called by ds_cont_tgt_ec_eph_query_ult() when ce_ref drops to zero.

Why don't we just call cont_ec_eph_destroy in ds_cont_ec_eph_delete() when ce_ref drops to zero?

Reply:
Because ds_cont_ec_eph_delete is called by ec aggregation ULT, which is per xstream,  and it will only drop the ref count of cont_ec_eph, 

when the ref count reach 0, i.e. all xstream delete its usage of this cont_ec_eph,  then the system xstream(0) will destroy cont_ec_eph in xstream(0) in ds_cont_tgt_ec_eph_query_ult().

If we call cont_ec_eph_destroy on other xstream, then it will need resolve the race between the list iteration in ds_cont_tgt_ec_eph_query_ult() and cont_ec_eph_destroy. 

So  ds_cont_tgt_ec_eph_query_ult() will just destroy the cont_ec_eph on its own to avoid the hassle.

Reply:
Given that ds_cont_ec_eph_delete() operates the global eph list and drops the ref count of global eph, it has to be called by sys xstream.

In current implementation, the ds_cont_ec_eph_delete() is called by sys xstream (see ec_agg_param_fini()), otherwise, it won't work at all.

Reply:
Yes, you are right about ds_cont_ec_eph_delete(), but iv_update might yield in that loop of ds_cont_tgt_ec_eph_query_ult(), so to avoid conflict between ds_cont_ec_eph_delete() and that loop, so I just let the loop to handle destroy().

Hmm, maybe I can do asynchronous IV update for eph update, but I do not much difference to handle delete by this system xstream ULT.

Reply:
Now I see the real cause by reading the ds_cont_tgt_ec_eph_query_ult() again, this ULT calls cont_iv_ec_agg_eph_update() which could yield in the loop of d_list_for_each_entry_safe(), so it'll be messed up if other ULT (on the same sys xstream) deleted item while it's yielding.

The problem could be solved by revise the implementation of ds_cont_tgt_ec_eph_query_ult(), but I'm good with current fix as well. Thanks.